clients/js/src/index.ts

```diff
@@ -0,0 +1,174 @@
+import { DefaultApi } from "./generated/api";
+import { Configuration } from "./generated/configuration";
+
+// a function to convert a non-Array object to an Array
+function toArray<T>(obj: T | Array<T>): Array<T> {
+  if (Array.isArray(obj)) {
+    return obj;
+  } else {
+    return [obj];
+  }
+}
+
+// a function to convert an array to array of arrays
+function toArrayOfArrays<T>(obj: Array<Array<T>> | Array<T>): Array<Array<T>> {
+  if (Array.isArray(obj[0])) {
+    return obj as Array<Array<T>>;
+  } else {
+    return [obj] as Array<Array<T>>;
+  }
+}
+
+export class Collection {
+  private name: string;
+  private api: DefaultApi;
+
+  constructor(name: string, api: DefaultApi) {
+    this.name = name;
+    this.api = api;
+  }
+
+  public async add(
```

Comment:
I think we need to think about validation as right now this doesn't seem to be doing any.

Right now the python client performs validation for this function in Collection.py, however that is skipped when going through the server. We could add validation here, make sure the server routes through Collection.py or both.

Reply:
It probably makes sense for us to have thin-clients and assume server will handle most validation.